### ISSUE TITLE
Integrating 0.1 Beta 2 through old way leads to errors

### ISSUE THREAD

**Issue by GoogleCodeExporter** [niltsh/panoramagl#2]
```
What steps will reproduce the problem?
1. Download and extract files to folder containing Project Directory
2. Follow steps at http://www.codeproject.com/KB/iPhone/panoramagl.aspx that 
instruct to Edit Active Target settings, add other target (for Linked 
Libraries), and then add the PanoramaGL Direct Dependency. (without forgetting 
to click on the checkbox after selecting the PanoramaGL.xcodeproj framework).
3. When attempting to run I get loaded with 75 parse errors.

What is the expected output? What do you see instead?

I expect it to work exactly like the previous version (at least it looked that 
way given the identical files).  Instead here are some of the parse errors I've 
been receiving:

These errors occur when compiling all of the various PLname.m files:

/Users/blahblah...blahblah/PanoramaGL/Classes/glues/glu.h:2:0 
/Users/blahblah...blahblah/PanoramaGL/Classes/glues/glu.h:2: error: expected 
'=', ',', ';', 'asm' or '__attribute__' before numeric constant

/Users/blahblah...blahblah/PanoramaGL/Classes/glues/glu.h:3:0 
/Users/blahblah...blahblah/PanoramaGL/Classes/glues/glu.h:3:1: error: invalid 
suffix "a7648d521f505b3b5b7948179679b73" on integer constant


What version of the product are you using? On what operating system?
PanoramaGL 0.1 Beta 2
OSX 10.6.4 on a 3.2 Apple SDK

Please provide any additional information below.
Not sure what to include, let me know and I'll post any info I can

```

Original issue reported on code.google.com by `Joe.Ekiert@gmail.com` on 7 Jul 2010 at 3:02

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
After just a little more investigating, it would appear my glu.h file is not 
what it probably should be.  Here are the contents:

XSym
0007
3a7648d521f505b3b5b7948179679b73
glues.h


I tried re-downloading the zip file and it looked to me like the file is still 
garbled.  I'm not quite sure if it is something on my end or if perhaps 
something occurred during zipping/uploading of the file.
```

Original comment by `Joe.Ekiert@gmail.com` on 7 Jul 2010 at 3:08

**Comment by GoogleCodeExporter** [comment]
```
that's right, i also found the same problem. by deleting glu.h and use glues.h 
instead. i can successfully compile the lib, but after i trying to compile 
hellopanorama, it failed again with a error says _objc_class_$_pltexture 
referenced from hellopanoramaviewcontroller.o symbols not found, collect2:id 
returned 1 exit status
```

Original comment by `sha...@gmail.com` on 24 Jul 2010 at 2:35

**Comment by GoogleCodeExporter** [comment]
```
I do believe that that is because it requires the glu header file - I attempted 
something similar where I used the glu.h file from the previous version.  
Unfortunately couldn't get the lib to compile.
```

Original comment by `Joe.Ekiert@gmail.com` on 26 Jul 2010 at 2:33

**Comment by GoogleCodeExporter** [comment]
```
just put #include <glues.h> in glu.h file and I can successfully compile the 
static lib. what i can't do is using the old way to actually use the PLView in 
hellopanorama project. it seems the hello project doesn't know where is the .h 
file
```

Original comment by `sha...@gmail.com` on 28 Jul 2010 at 6:07

**Comment by GoogleCodeExporter** [comment]
```
i guess they'd phased out the old glu.h file and to be compatible the old glu.h 
act like a pointer or sth. the real one is glues.h
```

Original comment by `sha...@gmail.com` on 28 Jul 2010 at 6:09

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
ok, I've been trying to email the author and he said it could be a ipad version 
in the future. he said sometimes earlier in Auguest.

Also after 2 weeks trying, now I've successfully compiled the whole thing and 
successfully run it on ipad. 2 tips for people who desperately trying to use 
this on ipad

1. modify the glu.h file, put #include "glues.h"
2. modify the panoramaGL project to include 2 other .h file in PL classes. 
include those 2 in the project. They already in the folder, just not included 
by default. I dont know if excluding them was on purpose but after I included 
them, everything is fine now.

```

Original comment by `sha...@gmail.com` on 7 Aug 2010 at 1:17

**Comment by GoogleCodeExporter** [comment]
```
I would like to use it in one of my projects, but I'm still getting compiling 
errors even after all those tips.

ld: duplicate symbol ___gl_pqHeapNewPriorityQ in 
/Users/hvilela/Documents/ConradCaine/Panorama/build/Panorama.build/Debug-iphones
imulator/Panorama.build/Objects-normal/i386/priorityq.o and 
/Users/hvilela/Documents/ConradCaine/Panorama/build/Panorama.build/Debug-iphones
imulator/Panorama.build/Objects-normal/i386/priorityq-heap.o

@shawhu Could you publish the fixed project somewhere?
```

Original comment by `henrique...@gmail.com` on 10 Aug 2010 at 7:22

**Comment by GoogleCodeExporter** [comment]
```
@henrique
do you use -all_load in "other linker flags"?
If yes, you have to change a few things in libtess.
```

Original comment by `christia...@gmail.com` on 29 Nov 2010 at 8:20

**Comment by GoogleCodeExporter** [comment]
```
@ christian
I get the same compiler error like henrique.. 
the reason for it is the -all_load flag.. I need that flag to compile the rest 
of my app.
Is there a way to compile the lib with the -all_load flag?
```

Original comment by `barpfa2...@googlemail.com` on 3 Dec 2010 at 9:09

**Comment by GoogleCodeExporter** [comment]
```
Maybe you could try using -force-load <lib path> instead of -all_load
```

Original comment by `attila.n...@gmail.com` on 3 Dec 2010 at 9:48

**Comment by GoogleCodeExporter** [comment]
```
I tried -force_load before, but it didn´t work,
Now I am back to -all-load and removed priorityq-heap.h and priorityq-heap.i 
from the target. 
It works now!
```

Original comment by `barpfa2...@googlemail.com` on 3 Dec 2010 at 10:50

**Comment by GoogleCodeExporter** [comment]
```
the problem relies on wrong include guards and the priorityq-heap.i file.
I've made a fixed version of panoramaGl and libtess and submited it to javier.
Hope he will publish it soon.
In the meantime you can use the attached one.
```

Original comment by `christia...@gmail.com` on 3 Dec 2010 at 3:18

Attachments:
- [libtess.zip](https://storage.googleapis.com/google-code-attachments/panoramagl/issue-2/comment-14/libtess.zip)

**Comment by GoogleCodeExporter** [comment]
```
@shawhu
2. modify the panoramaGL project to include 2 other .h file in PL classes. 
include those 2 in the project. They already in the folder, just not included 
by default. I dont know if excluding them was on purpose but after I included 
them, everything is fine now.

Could you please clarify what you have done exactly? Can you share the fixed 
project?
```

Original comment by `sots...@gmail.com` on 23 Feb 2011 at 10:30

**Comment by GoogleCodeExporter** [comment]
```
I need the fixed project too, im trying to run this on the ipad
Thanks!
```

Original comment by `gar...@gmail.com` on 6 Mar 2011 at 11:27

**Comment by GoogleCodeExporter** [comment]
```
is there anyone run this on ipad ? 


```

Original comment by `gereksiz...@gmail.com` on 17 Jun 2011 at 3:28

**Comment by GoogleCodeExporter** [comment]
```
yes ,i did run on ipad. got my hands dirty....
```

Original comment by `50866...@qq.com` on 28 Jun 2011 at 12:52

**Comment by GoogleCodeExporter** [comment]
```
Hi, can someone post a fully working example. I still get this ERROR:

Missing dependency target "PanoramaGL (from PanoramaGL.xcodeproj)"

Undefined symbols for architecture i386:
  "_OBJC_CLASS_$_PLResources", referenced from:
      objc-class-ref in libPanoramaGL.a(PLControl.o)
ld: symbol(s) not found for architecture i386
collect2: ld returned 1 exit status
```

Original comment by `meyb...@googlemail.com` on 12 Aug 2011 at 1:47

**Comment by GoogleCodeExporter** [comment]
```
Hey all,

I'm with a problem like the last one:
Undefined symbols for architecture i386:
  "_OBJC_CLASS_$_PLResources", referenced from:
      objc-class-ref in libPanoramaGL.a(PLControl.o)
ld: symbol(s) not found for architecture i386
collect2: ld returned 1 exit status

Can someone help me?

Thanks


```

Original comment by `jsin...@gmail.com` on 3 Jan 2012 at 6:59

**Comment by GoogleCodeExporter** [comment]
```
Hey man, I've downloaded a working version from this issue 1:"Problem exiting a 
PLView while scrolling"
The project name is: PanoramaGLPoC.zip 

Then try to match the settings between your project with the last panoramagl 
source version and the settings from the project PanoramaGLPoC.zip.

but i'm still can't implement the hotspot feature, if you have any clue about 
it, let's discuss about it.
```

Original comment by `Mumen...@gmail.com` on 3 Jan 2012 at 9:54

**Comment by GoogleCodeExporter** [comment]
```
Hey,

Thanks for your answer, In really I already use this library and wasn't so 
complicate to install. I think my problem is I'm using the new version of 
XCode, and by default it use the new Compiler LLVM 3.0. But this compiler I 
think have some problems with this library so I set the older LLVM GCC 4.2. 
When I try run this project (PanoramaGLPoC.zip) . The feedback i received is:

Command /Developer/Platforms/iPhoneOS.platform/Developer/usr/bin/llvm-gcc-4.2 
failed with exit code 1

Do you have same idea what can I do?

Thanks
```

Original comment by `jsin...@gmail.com` on 4 Jan 2012 at 1:05

**Comment by GoogleCodeExporter** [comment]
```
actually i've run it successfully and now trying to implement the hotspot 
feature.
```

Original comment by `Mumen...@gmail.com` on 4 Jan 2012 at 7:49

**Comment by GoogleCodeExporter** [comment]
```
Hi, 

Thanks for Mumen...@gmail.com, the code given by him worked fine.
Even I got the same error  "Command 
/Developer/Platforms/iPhoneOS.platform/Developer/usr/bin/llvm-gcc-4.2 failed 
with exit code 1"

What I did is I changed the compiler of the library and the projects compiler 
to LLVM GCC4.2.

Thanks once again guys :)

```

Original comment by `shravank...@gmail.com` on 5 Jan 2012 at 5:41

**Comment by GoogleCodeExporter** [comment]
```
Hi All,

  Worked here, I set the Valid Architecture to armv6 armv7 i386 and solved the problem.

@ Mumen.Sh, please when you have some news about the hotspot let me know.

Cheers
```

Original comment by `jsin...@gmail.com` on 5 Jan 2012 at 12:39

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
[deleted comment]
```

**Comment by GoogleCodeExporter** [comment]
```
hi to all,,, where can i download the latest version of PanoramaGL?? i have 
having a problem in memory leak

and the one that i download in codeproject.com having an error of this:


Apple LLVM compiler 3.1 Error
Method name referenced in property setter attribute must end with ':'


tnx man more power,

ill attach the file
```

Original comment by `rotski...@gmail.com` on 24 Apr 2012 at 2:33

Attachments:
- [PanoramaGL (2).zip](https://storage.googleapis.com/google-code-attachments/panoramagl/issue-2/comment-28/PanoramaGL %282%29.zip)

**Comment by GoogleCodeExporter** [comment]
```
@jsin: Yes that worked for me too.

I know that Hotspots aren't supported at the moment. Is there any other way to 
implement it. I haven't worked with openGL at all. But if it can be done using 
cocoa/uikit it would be intuitive to me(and others).

Also if someone knows how to get hotspots working, how will the transition 
between current to new hotspot be implemented?  Something like a motion blur 
effect.
```

Original comment by `gamergui...@gmail.com` on 9 May 2012 at 1:27

**Comment by GoogleCodeExporter** [comment]
```
@gamergui: could you point me a guide to use this library in xcode 4.3 and 
compile with ios5.0 sdk? Emm, similar to you, i don't have any ideas how OPENGL 
work. Hopefully I could see it's been done with UIKit. But first of all, how 
did you get library worked and successfully ran HelloPanorama?
```

Original comment by `JZhu.Wal...@gmail.com` on 13 May 2012 at 5:53

**Comment by GoogleCodeExporter** [comment]
```
All problems were fixed in 0.1 version, please check the last version, and wiki 
page, README file and last HelloPanoramaGL example
```

Original comment by `javbae...@gmail.com` on 5 Jul 2012 at 3:59
- Changed state: **Fixed**
